src/sqlfluff/core/templaters/slicers/tracer.py

```diff
@@ -70,9 +70,17 @@ def trace(self) -> JinjaTrace:
         trace_template = self.make_template(trace_template_str)
         trace_template_output = trace_template.render()
         # Split output by section. Each section has two possible formats.
-        for p in trace_template_output.split("\0"):
-            if not p:
-                continue
+        trace_entries = list(regex.finditer(r"\0", trace_template_output))
+        for match_idx, match in enumerate(trace_entries):
+            pos1 = match.span()[0]
+            try:
+                pos2 = trace_entries[match_idx + 1].span()[0]
+            except IndexError:
+                pos2 = len(trace_template_output)
+            p = trace_template_output[pos1 + 1 : pos2]
+            is_set = p[:3] == "set"
```

Comment:
This only covers the `{% set... %}` case but looking at https://jinja.palletsprojects.com/en/3.0.x/templates/ there are lots of similar cases (`for`, `call`, `macro`). Should we include them in this?

Or at least add a comment saying they are not supported for now?

Or where they already covered (I'd be really surprised if `for` was not covered already)?

Reply:
`{% for %}` is covered by the general block handling -- see the `block_types` lookup elsewhere in the file. Jinja is helping us with some of this by telling us the element type. TBH, I don't know if `call` and `macro` work. SInce I don't know for sure, I would rather not add comments about those.

Reply:
Note that `tracer.py` is the primary (only) component used in the past few releases for Jinja and dbt template slicing, so it's fairly robust. Given the extreme amount of templating used in dbt projects, I think it's pretty robust, just can't say for sure. AFAIK, this issue was the first one raised since we switched over in late October/early November.